interfaceUtils.py

```diff
@@ -40,127 +46,139 @@ def requestBuildArea(self):
 
         url = 'http://localhost:9000/blocks?x={}&y={}&z={}'.format(x, y, z)
-        # DEBUG: print(url)
         try:
             response = requests.get(url)
         except ConnectionError:
             return "minecraft:void_air"
         return response.text
-        # DEBUG: print("{}, {}, {}: {} - {}".format(x, y, z, response.status_code, response.text))
 
-    def setBlock(self, x, y, z, blockStr):
-        if self.useBatching:
-            self._placeBlockBatched(x, y, z, blockStr, self.batchSize)
+    def fill(self, x1, y1, z1, x2, y2, z2, str):
+        x1, y1, z1 = self.local2global(x1, y1, z1)
+        x2, y2, z2 = self.local2global(x2, y2, z2)
+        xlo, ylo, zlo = min(x1, x2), min(y1, y2), min(z1, z2)
+        xhi, yhi, zhi = max(x1, x2), max(y1, y2), max(z1, z2)
+
+        for x in range(xlo, xhi + 1):
+            for y in range(ylo, yhi + 1):
+                for z in range(zlo, zhi + 1):
+                    self.setBlock(x, y, z, str)
+
+    def setBlock(self, x, y, z, str):
+        """**Place a block in the world depending on buffer activation**."""
+        if self.__buffering:
+            self.placeBlockBatched(x, y, z, str, self.bufferlimit)
         else:
-            self._setBlockDirect(x, y, z, blockStr)
-
-    def sendOutstandingBlocks(self):
-        self._sendBlocks()
+            self.placeBlock(x, y, z, str)
 
-    def _setBlockDirect(self, x, y, z, blockStr):
-        """**Places a block in the world.**"""
+    def placeBlock(self, x, y, z, str):
+        """**Place a single block in the world**."""
         x, y, z = self.local2global(x, y, z)
 
         url = 'http://localhost:9000/blocks?x={}&y={}&z={}'.format(x, y, z)
-        # DEBUG: print('setting block {} at {} {} {}'.format(blockStr, x, y, z))
         try:
-            response = requests.put(url, blockStr)
+            response = requests.put(url, str)
         except ConnectionError:
             return "0"
         return response.text
-        # DEBUG: print("{}, {}, {}: {} - {}".format(x, y, z, response.status_code, response.text))
 
-    # --------------------------------------------------------- block buffers
+    # ----------------------------------------------------- block buffers
 
-    def _placeBlockBatched(self, x, y, z, blockStr, limit=50):
-        """**Place a block in the buffer and send if the limit is exceeded.**"""
+    def toggleBuffer(self):
+        """**Activates or deactivates the buffer function safely**."""
+        self.Buffering = not self.Buffering
+        return self.Buffering
+
+    @property
+    def Buffering(self):
+        return self.__buffering
+
+    @Buffering.setter
+    def Buffering(self, value):
+        self.__buffering = value
+        if self.__buffering:
+            print("Buffering has been activated.")
+        else:
+            self.sendBlocks()
+            print("Buffering has been deactivated.")
+
+    def placeBlockBatched(self, x, y, z, str, limit=50):
```

Comment:
Like with placeBlock, `str` should be kept as `blockStr` so it doesn't override the `str()` function.